stencils/LifecycleViewModel.stencil

```diff
@@ -7,12 +7,19 @@ class LifecycleViewModel<VM: BaseViewModel> {
     let containerView: ContainerView?
 
     init(_ viewModel: VM, containerView: ContainerView? = nil) {
+        assert(Thread.isMainThread, "Constructor must be called on the main thread, but called on \(Thread.current)")
```

Comment:
Is it something that was happening? Was init called not from the main thread?
I agree that we should initialize VM only from the main thread, but not sure that we should force it.

In fact, it is correct implementation if the class was initialized and disposed of in the same thread (not necessarily the main. 

If you had initialisation, not in main, then the problem may be that you were using the same but not the main queue. The global queue opposite to the main queue will not execute in the same thread all the time and it may cause the freezing issue.

Reply:
@dasdranagon the problem for me is to run the deinit cleanup on the same thread as init was. is there a way to ensure that?

Reply:
I don't think so